File: php-symfony/src/Entity/Access.php
<?php

namespace App\Entity;

use App\Repository\AccessRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: AccessRepository::class)]
class Access
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column]
    private ?\DateTimeImmutable $accessedAt = null;

    #[ORM\Column(nullable: true)]
    private ?array $userInfo = null;

    #[ORM\Column(type: Types::ARRAY, nullable: true)]
    private ?array $httpRequestHeaders = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getAccessedAt(): ?\DateTimeImmutable
    {
        return $this->accessedAt;
    }

    public function setAccessedAt(\DateTimeImmutable $accessedAt): static
    {
        $this->accessedAt = $accessedAt;

        return $this;
    }

    public function getUserInfo(): ?array
    {
        return $this->userInfo;
    }

    public function setUserInfo(?array $userInfo): static
    {
        $this->userInfo = $userInfo;

        return $this;
    }

    public function getHttpRequestHeaders(): ?array
    {
        return $this->httpRequestHeaders;
    }

    public function setHttpRequestHeaders(?array $httpRequestHeaders): static
    {
        $this->httpRequestHeaders = $httpRequestHeaders;

        return $this;
    }
}

File: java-spring/src/main/java/com/example/spring/access/entity/Access.java
package com.example.spring.access.entity;

import com.vladmihalcea.hibernate.type.json.JsonType;
import jakarta.persistence.*;
import org.hibernate.annotations.Type;

import java.sql.Timestamp;

@Entity
public class Access {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "accessed_at")
    private Timestamp accessedAt;

    @Type(JsonType.class)
    @Column(name = "user_info", columnDefinition = "json")
    private String userInfo;

    @Type(JsonType.class)
    @Column(name = "http_request_headers", columnDefinition = "json")
    private String httpRequestHeaders;

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public Timestamp getAccessedAt() {
        return accessedAt;
    }

    public void setAccessedAt(Timestamp accessedAt) {
        this.accessedAt = accessedAt;
    }

    public String getUserInfo() {
        return userInfo;
    }

    public void setUserInfo(String userInfo) {
        this.userInfo = userInfo;
    }

    public String getHttpRequestHeaders() {
        return httpRequestHeaders;
    }

    public void setHttpRequestHeaders(String httpRequestHeaders) {
        this.httpRequestHeaders = httpRequestHeaders;
    }
}
